LXReorderableCollectionViewFlowLayout/LXReorderableCollectionViewFlowLayout.m

```diff
@@ -138,7 +140,10 @@ - (void)applyLayoutAttributes:(UICollectionViewLayoutAttributes *)layoutAttribut
 }
 
 - (id<LXReorderableCollectionViewDataSource>)dataSource {
-    return (id<LXReorderableCollectionViewDataSource>)self.collectionView.dataSource;
+    if ( _dataSource == nil ) {
+        _dataSource = (id<LXReorderableCollectionViewDataSource>)self.collectionView.dataSource;
+    }
+    return _dataSource;
```

Comment:
There's no custom data source set within the library, but part of this feature was adding `dataSource` to the public interface. That way any code using this layout can set a custom data source, as opposed to just using the collectionview's datasource, as desired.

For example, in a project in which I'm using this, we are setting a view controller as the layout's data source. That way we can use this in conjunction with `RZCollectionListCollectionViewDataSource`.